### PR TITLE
Add warning to advanced settings

### PR DESCRIPTION
**Pull request by leonardola**
#### Changes proposed in this Pull Request
This PR adds a warning about incompatible extensions in the WooPay advanced settings page and fixes the icons being used all over the admin area

#### Testing instructions
- Checkout payments server to the latest version
- Change [this line](https://github.com/Automattic/woocommerce-payments-server/blob/trunk/server/wp-content/rest-api-plugins/endpoints/wcpay/woopay/class-woopay-controller.php#L60) to look like this:
`'incompatible_extensions' => ['test-extension', 'test-extension-2'],` 
- On your merchant site
- Execute the `validate_incompatible_extensions` cron
- Install and activate this extension
[test-extension.zip](https://github.com/Automattic/woocommerce-payments/files/11791369/test-extension.zip)
- Go to /wp-admin > Payments > Scroll to Express checkout
- Make sure there is a warning about incompatible extensions showing up
- Click the WooPay customize button
- Make sure there is a warning about incompatible extensions showing up
- Disable the extension and make sure the warning is gone in both places

The warnings should look like this:
**Settings page**
Screenshot 2023-07-05 at 5 56 28 PM
Screenshot 2023-07-05 at 5 56 44 PM

**Advanced Settings**
Screenshot 2023-07-05 at 5 56 01 PM
Screenshot 2023-07-05 at 5 55 55 PM

- Go to settings page
- Enable "Issue an authorization on checkout, and capture later"
- Make sure the payment methods look like this

Screenshot 2023-07-05 at 6 35 50 PM

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6691` or branch name `add/warning-to-advanced-settings` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/warning-to-advanced-settings) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 80fbc2ad4dfa40ba3ae63824a0d9c3a6d62cb27a
- Build time: 2023-07-14 17:25:38 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by leonardola** [comment]
@pierorocca @elizaan36 @stephendharmadi could you please take a look at this PR please? I've posted the screenshot for the changes in the description

**Comment by pierorocca** [comment]
@elizaan36 could you please comment on if the inline notification within the WooPay customization page is meant to be placed at the top of or below the setting?

**Comment by pierorocca** [comment]
@leonardola just saw your comment. Excuse me for asking the exact same thing. I think it needs to be consistently placed below to follow the notification system. I defer to @elizaan36. Let me grab the notification system guidelines for you.

**Comment by leonardola** [comment]
@pierorocca 
I've used the same standard of the other notifications in that page

Screenshot 2023-07-06 at 12 03 32 PM

**Comment by pierorocca** [comment]
Thanks for the visuals. It helps a lot. @elizaan36 @stephendharmadi reading through the shopper notices usage guideline, it's indeterminate if the placement of the notifications above fall within the guidance. The visual examples show the notification always near the top of a page or section. The written guidelines isn't as clear. Could you clarify please?

File: client/settings/settings-warnings/incompatibility-notice.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+					color: '#F0B849',
+					fill: 'currentColor',
+					marginBottom: '-5px',
+					marginRight: '10px',
+				} }
+				size={ 20 }
+			/>
+		</span>
+		<span>
+			{ __(
```
Comment (hsingyuc) [inline client/settings/settings-warnings/incompatibility-notice.js:32]: We are missing text domain 'woocommerce-payments' as a second argument here.

**Review by hsingyuc — COMMENTED** [review]
@leonardola Thank you for working on this!  Tests as expected, and I left one comment below.

**Comment by stephendharmadi** [comment]
I know Elizabeth designed these notification guidelines, but I would say that the notification banner should appear at the top of configurations. The rationale is to let merchants know that any changes would affect the other buttons before making any changes in settings.

**Comment by elizaan36** [comment]
Sorry for the delay in response.
+1 to Stephens comment. The notifications can be consistently placed above the relevant setting.

File: client/settings/settings-warnings/incompatibility-notice.js (hunk 1 above)
Comment (leonardola) [reply client/settings/settings-warnings/incompatibility-notice.js:32]: Fixed with 9d010d60035e392a504f252b45f43ff73e242ce6

**Review by hsingyuc — APPROVED** [review]
Tested again, as expected, and LGTM. 👍

**Pull request merged**